### PR TITLE
[YONK-1241] [MCKIN-8418] [MCKIN-8730] Social engagement optimizations

### PR DESCRIPTION
**Pull request by Agrendalath**
This reverts #1302 and adds changes from #1287.

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @Agrendalath! I've created [OSPR-2903](https://openedx.atlassian.net/browse/OSPR-2903) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

File: requirements/edx/custom.txt [hunk 1]
```diff
@@ -22,11 +22,11 @@ git+https://github.com/edx-solutions/xblock-group-project.git@0.1.1#egg=xblock-g
 -e git+https://github.com/mckinseyacademy/xblock-diagnosticfeedback.git@v0.2.4#egg=xblock-diagnostic-feedback==0.2.4
 -e git+https://github.com/open-craft/xblock-group-project-v2.git@0.4.10#egg=xblock-group-project-v2==0.4.10
 -e git+https://github.com/open-craft/xblock-virtualreality.git@v0.1.1#egg=xblock-virtualreality==0.1.1
-git+https://github.com/edx-solutions/api-integration.git@v2.3.14#egg=api-integration==2.3.14
+git+https://github.com/edx-solutions/api-integration.git@agrendalath/revert-revert-MCKIN-8418-MCKIN-8730#egg=api-integration==2.3.15
```
Comment (naeem91) [inline requirements/edx/custom.txt]: Since https://github.com/edx-solutions/api-integration/pull/185 is being merged, we can use the release tag here?

**Comment by naeem91** [comment]
@Agrendalath Will thumb when updated tag (of https://github.com/edx-solutions/api-integration/pull/185) is used build is green

File: requirements/edx/custom.txt (hunk 1 above)
Comment (Agrendalath) [reply requirements/edx/custom.txt]: I included the branch here to test `api-integration` before merging. I will push a new commit here with a proper version after merging `api-integration`.

Comment (naeem91) [reply requirements/edx/custom.txt]: ah, ok then will thumb up when tests are passed

**Pull request merged**